inference-engine/src/transformations/src/transformations/op_conversions/fq_decomposition.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include "transformations/op_conversions/fq_decomposition.hpp"
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/builder/autobroadcast.hpp>
+
+#include <numeric>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::FakeQuantizeDecomposition, "FakeQuantizeDecomposition", 0);
+
+bool isValidRangesInputs(const std::shared_ptr<ngraph::opset1::FakeQuantize> &fq) {
+    for (size_t i = 1; i < fq->get_input_size(); i++) {
+        if (!std::dynamic_pointer_cast<const ngraph::opset1::Constant>(fq->get_input_node_shared_ptr(i)))
+            return false;
+    }
+
+    auto il_node = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(fq->get_input_node_shared_ptr(1));
+    auto ih_node = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(fq->get_input_node_shared_ptr(2));
+
+    auto broadcasted_shape = ngraph::builder::get_numpy_broadcast_shapes({il_node->get_shape(), ih_node->get_shape()});
+    auto result_shape = broadcasted_shape.first;
+    auto il_padded_shape = broadcasted_shape.second.at(0);
+    auto ih_padded_shape = broadcasted_shape.second.at(1);
+
+    auto get_strides = [](const ngraph::Shape& in_shape, const ngraph::Shape& res_shape) -> std::vector<size_t> {
+        std::vector<size_t> strides(res_shape.size());
+        size_t k = 1;
+        for (int i = in_shape.size() - 1; i >= 0; i--) {
+            strides[i] = (in_shape[i] == res_shape[i]) ? k : 0;
+            k *= in_shape[i];
+        }
+        return strides;
+    };
+
+    auto il_strides = get_strides(il_padded_shape, result_shape);
+    auto ih_strides = get_strides(ih_padded_shape, result_shape);
+
+    auto get_index = [](const ngraph::Coordinate& in_coord, const std::vector<size_t>& strides) -> size_t {
+        size_t index = 0;
+        for (size_t i = 0; i < in_coord.size(); i++) {
+            index += in_coord[i]*strides[i];
+        }
+        return index;
+    };
+
+    const std::vector<float> il = il_node->cast_vector<float>();
+    const std::vector<float> ih = ih_node->cast_vector<float>();
+
+    auto step = [](ngraph::Coordinate& _iter, const ngraph::Shape& _dims) {
+        auto iter = _iter.rbegin();
+        auto dims = _dims.rbegin();
+
+        while (iter != _iter.rend()) {
+            *iter = (*iter + 1) % *dims;
+            if (*iter != 0) {
+                break;
+            }
+            ++iter;
+            ++dims;
+        }
+    };
+
+    const size_t work_amount = std::accumulate(result_shape.begin(), result_shape.end(), 1, std::multiplies<size_t>());
+    ngraph::Coordinate iter(result_shape.size(), 0);
+    for (size_t i = 0; i < work_amount; i++) {
+        if (il[get_index(iter, il_strides)] >= ih[get_index(iter, ih_strides)])
+            return false;
+        step(iter, result_shape);
+    }
```

Comment:
@GlebKazantaev we have a necessity to compare two tensors (respecting broadcasting) and if at least one element of the first tensor is bigger then the respective element form the second one we should disable transformation. It the current approach the best way to implement it (maybe we have some auxiliary stuff for these tasks)? Or ideologically it is better to create subgraph with GreaterEqual operation and evaluate it inplace (after we just need to analyze evaluate result)?

Reply:
Discussed offline
Was decided to use ```constant_fold``` for ```GreaterEqual```